### PR TITLE
Improved performances for assignment listing

### PR DESCRIPTION
**Pull request by giocalitri**
- Modified "staff_grading_data" function to be able to pull infos for
many students or a specific one
- Fixed bug with multiple identical incremental ajax requests:
- Now the cancel button of the "Enter Grade" modal does not trigger an
Ajax get request

### CODE REVIEW

**Comment by pdpinch** [comment]
This works and looks good to me. It needs JavaScript tests, but I suppose that will have to come after #127 is merged and #113 is worked on.

File: edx_sga/tests.py [hunk 1]
```diff
@@ -564,6 +564,36 @@ def test_get_staff_grading_data(self):
...
+            annotated_filename="foo_corrected.txt",
+            comment="Good work!")['module']
+        fred = self.make_student(
+            block, "fred",
+            filename="bar.txt")['module']
+        data = block.staff_grading_data()
+        assignments = data.get('assignments')
+        self.assertIsNotNone(assignments)
+        self.assertEqual(len(assignments), 2)
+        # 50b287502d8dbf83bec2dcfb78fc4d8e
```
Comment (noisecapella) [inline edx_sga/tests.py]: What's this for?

**Comment by pdpinch** [comment]
Oy. I forgot I need to run tests by hand, and they're not passing...

File: edx_sga/tests.py (hunk 1 above)
Comment (giocalitri) [inline edx_sga/tests.py]: I left that behind: it's the anonymous student ID for that test

**Comment by giocalitri** [comment]
Yes, I suggest to revert this merge and reopen the PR: we need to figure out if they fail also on master (like it looked to me) or only after my changes.

**Comment by pdpinch** [comment]
Right, right. You told me about that. I forgot. I'll revert it.

**Comment by giocalitri** [comment]
If you restore the branch, I'll remove that useless comment @noisecapella found

**Comment by pdpinch** [comment]
branch is restored. 

And I have a PR with the revert.

**Pull request merged**